packages/js/src/utils/errors.ts

```diff
@@ -32,15 +32,43 @@ const getMessageByErrorCode = (errorCode: LogtoErrorCode): string => {
   return errorCode;
 };
 
+interface LogtoErrorParameters {
+  code: LogtoErrorCode;
+  error?: string;
+  errorDescription?: string;
+  message?: string;
+}
+
 export class LogtoError extends Error {
   code: LogtoErrorCode;
   error?: string;
   errorDescription?: string;
 
-  constructor(code: LogtoErrorCode, error?: string, errorDescription?: string, message?: string) {
+  constructor(payload: LogtoErrorCode | LogtoErrorParameters) {
+    if (typeof payload === 'string') {
+      super(getMessageByErrorCode(payload));
+      this.code = payload;
+      return;
+    }
+
+    const { code, error, errorDescription, message } = payload;
     super(message ?? getMessageByErrorCode(code));
     this.code = code;
     this.error = error;
     this.errorDescription = errorDescription;
   }
 }
+
+export interface LogtoRequestErrorParameters {
+  code?: string;
```

Comment:
should it be required?

Reply:
Updated, PTAL.

Reply:
(Deleted a duplicated comment below.)

Reply:
Outdated, follow the new implementation.